h/services/group_members.py

```diff
@@ -104,15 +104,14 @@ def update_members(self, group, userids):
                         be the members of this group
         """
         current_mem_ids = [member.userid for member in group.members]
-        userids_for_removal = [
-            mem_id for mem_id in current_mem_ids if mem_id not in userids
-        ]
 
         for userid in userids:
-            self.member_join(group, userid)
+            if userid not in current_mem_ids:
+                self.member_join(group, userid)
```

Comment:
🤷‍♂️ If the member already exists, there's no need to add them. This `if` is the entire fix. Can it really be so simple? And why wasn't the code written like this before?